File: AlgorithmsPackJavaSolutions/976. Largest Perimeter Triangle.java
public class Solution {
    public int largestPerimeter(int[] nums) {
        int sz = nums.length;
        Arrays.sort(nums);
        for (int i = sz - 1; i >= 2; i--) {
            if (nums[i - 2] + nums[i - 1] > nums[i]) {
                return nums[i - 2] + nums[i - 1] + nums[i];
            }
        }
        return 0;
    }
}

File: AlgorithmsPackPythonSolutions/976. Largest Perimeter Triangle.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        sz = len(nums)
        nums.sort(reverse=True)
        for i in range(sz - 2):
            if nums[i] < nums[i + 1] + nums[i + 2]:
                return nums[i] + nums[i + 1] + nums[i + 2]
        return 0
